src/cpp/session/SessionModuleContext.cpp

```diff
@@ -1817,7 +1820,7 @@ bool fileListingFilter(const core::FileInfo& fileInfo)
    {
       return true;
    }
-   else if (userSettings().hideObjectFiles() &&
+   else if (prefs::userPrefs().hideObjectFiles() &&
```

Comment:
Note: I recall from a previous PR that we wanted to move this logic outside of this function, as this gets called in the file monitor thread and can lead to race conditions in access. Is that handled now? (IIUC access is now guarded by a mutex and so this should be okay?)

Reply:
Yes, we now guard all pref reads/writes with a mutex!